microsoft/testsuites/dpdk/common.py

```diff
@@ -17,6 +22,260 @@
 AZ_ROUTE_ALL_TRAFFIC = "0.0.0.0/0"
 
 
+# Attempt to clean up the DPDK package dependency mess
+# Make a Installer class that implements the common steps
+# for installing DPDK/rdma-core, either from source or the package manager.
+# This generic class will get implemented in DpdkTestpmd and RdmaCore.
+# This should help us cover the various installation cases in a nice way,
+# and allow us to only re-implement the bits we need for each project.
+class OsPackageDependencies:
+    # A class to reduce the isinstance() trees that are
+    # sprinkled everywhere.
+    # Caller provides a function to match an OS and
+    # the packages to install on that OS.
+    def __init__(
+        self,
+        matcher: Callable[[Posix], bool],
+        packages: Optional[Sequence[Union[str, Tool, Type[Tool]]]] = None,
+        exclusive_match: bool = False,
+    ) -> None:
+        self.matcher = matcher
+        self.packages = packages
+        self.exclusive_match = exclusive_match
+
+
+class DependencyInstaller:
+    # provide a list of OsPackageDependencies for a project
+    def __init__(self, requirements: List[OsPackageDependencies]) -> None:
+        self.requirements = requirements
+
+    # evaluate the list of package dependencies,
+    def install_required_packages(
+        self, node: Node, extra_args: Union[List[str], None]
+    ) -> None:
+        os = node.os
+        if not isinstance(os, Posix):
+            return
+        # find the match for an OS, install the packages.
+        # stop on list end or if exclusive_match parameter is true.
```

Comment:
as the comments, call the variable `stop_proceed` or similar. The `exclusive_match` sounds not ordered, but actually those rules are ordered, and handle one by one.